src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass_Switch.cs

```diff
@@ -182,11 +188,17 @@ private void VisitDecisionTree(DecisionTree decisionTree)
                 case DecisionTree.DecisionKind.ByValue:
                     {
                         var byValue = (DecisionTree.ByValue)decisionTree;
+                        var expressionIsConstant = byValue.Expression.ConstantValue != null;
+                        bool valueHandled = false;
                         foreach (var kvp in byValue.ValueAndDecision)
                         {
-                            VisitDecisionTree(kvp.Value);
+                            if (!expressionIsConstant || Equals(byValue.Expression.ConstantValue.Value, kvp.Key))
```

Comment:
Consider extracting a local for `byValue.Expression.ConstantValue`. And then this condition might be easier to read as: `if (constValue == null || Equals(constValue.Value, kvp.Key))`.